File: src/Luna/Codegen/Hs/Path.hs
---------------------------------------------------------------------------
-- Copyright (C) Flowbox, Inc - All Rights Reserved
-- Unauthorized copying of this file, via any medium is strictly prohibited
-- Proprietary and confidential
-- Flowbox Team <contact@flowbox.io>, 2013
---------------------------------------------------------------------------

module Luna.Codegen.Hs.Path (
    module Luna.Network.Path.Path,
    toModulePath,
    toModulePaths
)where

import           Luna.Network.Path.Path 
import           Data.Char                      (isLower)
import           Data.String.Utils                 (join)


toModulePath :: Path -> String
toModulePath path = join "." (toModulePaths path)


toModulePaths :: Path -> [String]
toModulePaths path = case path of
    Path []                -> []
    Path [segment@(x:xs)]  -> [name] where
                                  name = if isLower x
                                      then "U'" ++ segment
                                      else segment
    Path (x:xs)            -> toModulePaths (Path [x]) ++ toModulePaths (Path xs)

File: scripts/utils/path.py
###########################################################################
## Copyright (C) Flowbox, Inc / All Rights Reserved
## Unauthorized copying of this file, via any medium is strictly prohibited
## Proprietary and confidential
## Flowbox Team <contact@flowbox.io>, 2014
###########################################################################

import os
import tarfile

def write_if_changed(path, s):
    if os.path.exists(path):
        with open(path, 'r') as file:
            if file.read() == s:
                return
    with open(path, 'w') as file:
        file.write(s)

def make_dirs_if_needed(path):
    if not os.path.exists(os.path.dirname(path)):
        os.makedirs(path)


def extract(rootdir, path):
    tar = tarfile.open(path)
    for tarinfo in tar:
        path = os.path.join(rootdir, tarinfo.name)
        if os.path.exists(path):
            print_info("File %s already exists. Skipping extraction." % path)
            break
        tar.extractall(rootdir, [tarinfo])